### PR TITLE
feat: storing cart attributes as cookie

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

>  We will be storing the cart attributes in cookies.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: deviceModeInit.js [hunk 1]
```diff
@@ -153,32 +159,46 @@ var rudderTracking = (function () {
...
-      return true;
+    if (attributes?.rudderAnonymousId) {
+      return false;
     }
-    return false;
+    return true;
   }
 
+  function updateCookies() {
+    const expiryInDays = 14;
```
Comment (ItsSudip) [inline deviceModeInit.js:169]: Let's keep the expiry for 2 hours only. After each hour we will update it.

File: deviceModeInit.js [hunk 2]
```diff
@@ -153,32 +159,46 @@ var rudderTracking = (function () {
...
+      const keyVal = cookie.split("=");
+      const name = keyVal[0];
+      const value = keyVal[1];
+      if (name === "rs_shopify_updated_at") {
+        const oneHourTimeInMilliSeconds = 60 * 60 * 1000;
+        const currentTime = Date.now();
+        return currentTime - value > oneHourTimeInMilliSeconds;
+      }
+    });
+    return true;
```
Comment (utsabc) [inline deviceModeInit.js:201]: There is already a cookie utility below in you can use that

File: deviceModeInit.js [hunk 3]
```diff
@@ -153,32 +159,46 @@ var rudderTracking = (function () {
...
-    return false;
+    return true;
   }
 
+  function updateCookies() {
+    const expiryInHours = 2;
+    const date = new Date();
+    date.setTime(date.getTime() + (expiryInHours * 60 * 60 * 1000));
+    const expires = "expires=" + date.toUTCString();
+    document.cookie = `rs_shopify_updated_at = ${Date.now()}; ${expires};`;
```
Comment (utsabc) [inline deviceModeInit.js:173]: Use the cookie utility below to set it

**Review by utsabc — CHANGES_REQUESTED** [review]

File: deviceModeInit.js (hunk 1 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

File: deviceModeInit.js (hunk 2 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

File: deviceModeInit.js (hunk 3 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

File: deviceModeInit.js [hunk 4]
```diff
@@ -153,32 +159,40 @@ var rudderTracking = (function () {
...
   }
-
-  function checkUpdateTime(cart) {
+  function checkUpdateTime() {
     const oneHourTimeInMilliSeconds = 60 * 60 * 1000;
-    const { rudderUpdatedAt } = cart.attributes;
     const currentTime = Date.now();
-    return currentTime - rudderUpdatedAt > oneHourTimeInMilliSeconds;
+    const cookies = cookie_parse();
+    const lastupdatedValue = cookies?.rs_shopify_updated_at || 0;
```
Comment (utsabc) [inline deviceModeInit.js:194]: Use `cookie_action` with `get` parameter to read cookies check line 142

**Review by utsabc — CHANGES_REQUESTED** [review]

File: deviceModeInit.js [hunk 5]
```diff
@@ -153,32 +159,40 @@ var rudderTracking = (function () {
...
-    if (attributes?.rudderAnonymousId && attributes?.rudderUpdatedAt) {
-      return true;
+    if (attributes?.rudderAnonymousId) {
+      return false;
     }
-    return false;
+    return true;
   }
 
+  function updateCookies() {
```
Comment (utsabc) [inline deviceModeInit.js:168]:
```suggestion
  function updateTimeStampFoIdentifierEvent() {
```

File: deviceModeInit.js [hunk 6]
```diff
@@ -153,32 +159,40 @@ var rudderTracking = (function () {
...
+    const cookieOptions = {
+      action: "set",
+      expire_hr: 2,
+      name: "rs_shopify_updated_at",
```
Comment (utsabc) [inline deviceModeInit.js:172]:
```suggestion
      name: "rs_shopify_cart_identified_at",
```

**Review by utsabc — CHANGES_REQUESTED** [review]

File: deviceModeInit.js (hunk 4 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

File: deviceModeInit.js (hunk 5 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

File: deviceModeInit.js (hunk 6 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

File: deviceModeInit.js [hunk 7]
```diff
@@ -85,11 +85,17 @@ var rudderTracking = (function () {
         : "";
     fetchCart()
       .then((cart) => {
-        if (!checkCartAttributes(cart) || checkUpdateTime(cart)) {
+        const needToUpdateCart = checkCartAttributes(cart);
```
Comment (utsabc) [inline deviceModeInit.js:88]: The function name `checkCartAttributes` does not match with behaviour change either name  or behaviour

File: deviceModeInit.js [hunk 8]
```diff
@@ -153,32 +159,43 @@ var rudderTracking = (function () {
...
-          rudderUpdatedAt: Date.now(),
         },
       },
       undefined,
       "json"
     );
   }
-
-  function checkUpdateTime(cart) {
+  function checkUpdateTime() {
```
Comment (utsabc) [inline deviceModeInit.js:190]: Function name is not synonymous to behaviour please address

File: deviceModeInit.js (hunk 7 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

File: deviceModeInit.js (hunk 8 above)
Comment (anantjain45823) [reply deviceModeInit.js]: Done

**Review by utsabc — APPROVED** [review]

**Pull request merged**